client/src/Models/Ringdown.js

```diff
@@ -7,20 +7,70 @@ import * as metadata from '../../../shared/metadata';
 import convertToPropType from '../utils/convertToPropType';
 import DeliveryStatus from '../../../shared/constants/DeliveryStatus';
 
+// import ambulance from '../../../shared/metadata/ambulance';
+// import patient from '../../../shared/metadata/patient';
+
+const Patient = metadata.patient.getFieldHash();
+const Ambulance = metadata.ambulance.getFieldHash();
+const EmergencyMedicalServiceCall = metadata.emergencyMedicalServiceCall.getFieldHash();
+
 // define the fields that must all have valid input to make the ringdown valid.  the second array item is an optional function to determine
 // whether the field's current value is valid as input.  by default, the field is counted as having input if its value is truthy.   the
 // array order should be the same as the field order in PatientFields.
+
+const handleInputValidation = (name, value) => {
+  const { type, required = false, allowNull = true } = Patient[name] || Ambulance[name] || EmergencyMedicalServiceCall[name] || {};
+
+  let isValidType;
+  switch (type) {
+    case 'integer':
+      isValidType = typeof value === 'number' && value % 1 === 0;
+      break;
+    case 'boolean':
+      isValidType = typeof value === 'boolean';
+      break;
+    case 'enum':
+    case 'text':
+      isValidType = typeof value === 'string';
+      break;
+    case 'decimal':
+      isValidType = typeof value === 'number';
+      break;
+    default:
+      break;
+  }
+
+  return { value, isRequired: required, allowNull, isValidType };
+};
 const validatedFields = [
-  ['ambulanceIdentifier'],
-  ['dispatchCallNumber'],
-  ['emergencyServiceResponseType'],
-  ['age'],
-  ['sex'],
-  ['chiefComplaintDescription'],
-  ['stableIndicator', (value) => typeof value === 'boolean'],
-  ['all', () => ValidationState.NO_INPUT],
+  'ambulanceIdentifier',
+  'dispatchCallNumber',
+  'emergencyServiceResponseType',
+  'age',
+  'sex',
+  'chiefComplaintDescription',
+  'stableIndicator',
+  'systolicBloodPressure',
+  'diastolicBloodPressure',
+  'heartRateBpm',
+  'respiratoryRate',
+  'oxygenSaturation',
+  'supplementalOxygenAmount',
+  'temperature',
+  'glasgowComaScale',
+  'all',
 ];
 
+const handleRange = (value, max = null, min = null) => {
+  if (isNaN(min)) {
+    return value && value > max ? false : true;
```

Comment:
This would return false for 0, right?  If there's no min, I'd assume anything less than max is valid, which would include 0.